test/AutoDiff/validation-test/forward_mode.swift

```diff
@@ -1319,4 +1319,23 @@ ForwardModeTests.test("ForceUnwrapping") {
   expectEqual(5, forceUnwrap(Float(2)))
 }
 
+//===----------------------------------------------------------------------===//
+// Array methods from ArrayDifferentiation.swift
+//===----------------------------------------------------------------------===//
+
+ForwardModeTests.test("Array.differentiableMap") {
+  let a: [Float] = [1, 2, 3]
+  let tan = Array<Float>.TangentVector.init(a)
+
+  func multiplyMap(_ a: [Float]) -> [Float] {
+    return a.differentiableMap({ x in 3 * x })
+  }
+  expectEqual([3, 3, 3], differential(at: [1, 1, 1], in: multiplyMap)(tan))
```

Comment:
> Not sure if the test is passing now -- `lit` fails with ` Aborted (core dumped) ... a.out` at some point for me)

Here's the `lit` output I got for this test:

<details>

```console
$ llvm-project/llvm/utils/lit/lit.py -vvv build/Ninja-ReleaseAssert/swift-macosx-x86_64/test-macosx-x86_64/AutoDiff --filter forward_mode

...

FAIL: Swift(macosx-x86_64) :: AutoDiff/validation-test/forward_mode.swift (103 of 105)
******************** TEST 'Swift(macosx-x86_64) :: AutoDiff/validation-test/forward_mode.swift' FAILED ********************
Script:
--
: 'RUN: at line 1';   rm -rf "/Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/test-macosx-x86_64/AutoDiff/validation-test/Output/forward_mode.swift.tmp" && mkdir -p "/Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/test-macosx-x86_64/AutoDiff/validation-test/Output/forward_mode.swift.tmp" && xcrun --toolchain default --sdk '/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.16.sdk' /Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/bin/swiftc -target x86_64-apple-macosx10.9  -module-cache-path '/Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/swift-test-results/x86_64-apple-macosx10.9/clang-module-cache'  -F '/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks' -toolchain-stdlib-rpath -Xlinker -rpath -Xlinker '/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks' -Xlinker -rpath -Xlinker /usr/lib/swift  -swift-version 4  -Xfrontend -ignore-module-source-info  -F '/Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/lib' -Xlinker -rpath -Xlinker '/Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/lib' -module-cache-path '/Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/swift-test-results/x86_64-apple-macosx10.9/clang-module-cache' /Users/danielzheng/swift-build/swift/test/AutoDiff/validation-test/forward_mode.swift -Xfrontend -enable-experimental-forward-mode-differentiation -o /Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/test-macosx-x86_64/AutoDiff/validation-test/Output/forward_mode.swift.tmp/a.out -module-name main  && codesign -f -s - /Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/test-macosx-x86_64/AutoDiff/validation-test/Output/forward_mode.swift.tmp/a.out && /usr/bin/env DYLD_LIBRARY_PATH='/Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/lib/swift/macosx' LD_LIBRARY_PATH='/Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/lib/swift/macosx:' SIMCTL_CHILD_DYLD_LIBRARY_PATH='/Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/lib/swift/macosx'  /Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/test-macosx-x86_64/AutoDiff/validation-test/Output/forward_mode.swift.tmp/a.out
--
Exit Code: 134

Command Output (stdout):
--
[ RUN      ] ForwardModeDifferentiation.Identity
[       OK ] ForwardModeDifferentiation.Identity
[ RUN      ] ForwardModeDifferentiation.Unary
[       OK ] ForwardModeDifferentiation.Unary
[ RUN      ] ForwardModeDifferentiation.Binary
[       OK ] ForwardModeDifferentiation.Binary
[ RUN      ] ForwardModeDifferentiation.BinaryWithLets
[       OK ] ForwardModeDifferentiation.BinaryWithLets
[ RUN      ] ForwardModeDifferentiation.SubsetParametersDiff
[       OK ] ForwardModeDifferentiation.SubsetParametersDiff
[ RUN      ] ForwardModeDifferentiation.UnaryWithVars
[       OK ] ForwardModeDifferentiation.UnaryWithVars
[ RUN      ] ForwardModeDifferentiation.StructInit
[       OK ] ForwardModeDifferentiation.StructInit
[ RUN      ] ForwardModeDifferentiation.StructExtract
[       OK ] ForwardModeDifferentiation.StructExtract
[ RUN      ] ForwardModeDifferentiation.LocalStructVariable
[       OK ] ForwardModeDifferentiation.LocalStructVariable
[ RUN      ] ForwardModeDifferentiation.noParamMethodA
[       OK ] ForwardModeDifferentiation.noParamMethodA
[ RUN      ] ForwardModeDifferentiation.noParamMethodx
[       OK ] ForwardModeDifferentiation.noParamMethodx
[ RUN      ] ForwardModeDifferentiation.complexBinaryMethod
[       OK ] ForwardModeDifferentiation.complexBinaryMethod
[ RUN      ] ForwardModeDifferentiation.TrackedIdentity
[       OK ] ForwardModeDifferentiation.TrackedIdentity
[ RUN      ] ForwardModeDifferentiation.TrackedAddition
[       OK ] ForwardModeDifferentiation.TrackedAddition
[ RUN      ] ForwardModeDifferentiation.TrackedDivision
[       OK ] ForwardModeDifferentiation.TrackedDivision
[ RUN      ] ForwardModeDifferentiation.TrackedMultipleMultiplication
[       OK ] ForwardModeDifferentiation.TrackedMultipleMultiplication
[ RUN      ] ForwardModeDifferentiation.TrackedWithLets
[       OK ] ForwardModeDifferentiation.TrackedWithLets
[ RUN      ] ForwardModeDifferentiation.TupleLet
[       OK ] ForwardModeDifferentiation.TupleLet
[ RUN      ] ForwardModeDifferentiation.TupleVar
[       OK ] ForwardModeDifferentiation.TupleVar
[ RUN      ] ForwardModeDifferentiation.TupleMutation
[       OK ] ForwardModeDifferentiation.TupleMutation
[ RUN      ] ForwardModeDifferentiation.TupleNonDifferentiableElements
[       OK ] ForwardModeDifferentiation.TupleNonDifferentiableElements
[ RUN      ] ForwardModeDifferentiation.GenericIdentity
[       OK ] ForwardModeDifferentiation.GenericIdentity
[ RUN      ] ForwardModeDifferentiation.GenericTensorIdentity
[       OK ] ForwardModeDifferentiation.GenericTensorIdentity
[ RUN      ] ForwardModeDifferentiation.GenericTensorPlus
[       OK ] ForwardModeDifferentiation.GenericTensorPlus
[ RUN      ] ForwardModeDifferentiation.GenericTensorBinaryInput
[       OK ] ForwardModeDifferentiation.GenericTensorBinaryInput
[ RUN      ] ForwardModeDifferentiation.GenericTensorWithLets
[       OK ] ForwardModeDifferentiation.GenericTensorWithLets
[ RUN      ] ForwardModeDifferentiation.GenericTensorWithVars
[       OK ] ForwardModeDifferentiation.GenericTensorWithVars
[ RUN      ] ForwardModeDifferentiation.GenericTrackedIdentity
[       OK ] ForwardModeDifferentiation.GenericTrackedIdentity
[ RUN      ] ForwardModeDifferentiation.GenericTrackedBinaryAdd
[       OK ] ForwardModeDifferentiation.GenericTrackedBinaryAdd
[ RUN      ] ForwardModeDifferentiation.GenericTrackedBinaryLets
[       OK ] ForwardModeDifferentiation.GenericTrackedBinaryLets
[ RUN      ] ForwardModeDifferentiation.GenericTrackedBinaryVars
[       OK ] ForwardModeDifferentiation.GenericTrackedBinaryVars
[ RUN      ] ForwardModeDifferentiation.TrackedDifferentiableFuncType
[       OK ] ForwardModeDifferentiation.TrackedDifferentiableFuncType
[ RUN      ] ForwardModeDifferentiation.Final
[       OK ] ForwardModeDifferentiation.Final
[ RUN      ] ForwardModeDifferentiation.Simple
[       OK ] ForwardModeDifferentiation.Simple
[ RUN      ] ForwardModeDifferentiation.SimpleWrtSelf
[       OK ] ForwardModeDifferentiation.SimpleWrtSelf
[ RUN      ] ForwardModeDifferentiation.Simple Protocol
[       OK ] ForwardModeDifferentiation.Simple Protocol
[ RUN      ] ForwardModeDifferentiation.ProtocolFunc
[       OK ] ForwardModeDifferentiation.ProtocolFunc
[ RUN      ] ForwardModeDifferentiation.constructor, accessor, subscript
[       OK ] ForwardModeDifferentiation.constructor, accessor, subscript
[ RUN      ] ForwardModeDifferentiation.Arithmetics
[       OK ] ForwardModeDifferentiation.Arithmetics
[ RUN      ] ForwardModeDifferentiation.Fanout
[       OK ] ForwardModeDifferentiation.Fanout
[ RUN      ] ForwardModeDifferentiation.FunctionCall
[       OK ] ForwardModeDifferentiation.FunctionCall
[ RUN      ] ForwardModeDifferentiation.ResultSelection
[       OK ] ForwardModeDifferentiation.ResultSelection
[ RUN      ] ForwardModeDifferentiation.CaptureLocal
[       OK ] ForwardModeDifferentiation.CaptureLocal
[ RUN      ] ForwardModeDifferentiation.CaptureGlobal
[       OK ] ForwardModeDifferentiation.CaptureGlobal
[ RUN      ] ForwardModeDifferentiation.Mutation
[       OK ] ForwardModeDifferentiation.Mutation
[ RUN      ] ForwardModeDifferentiation.StructMemberwiseInitializer
[       OK ] ForwardModeDifferentiation.StructMemberwiseInitializer
[ RUN      ] ForwardModeDifferentiation.StructConstantStoredProperty
[       OK ] ForwardModeDifferentiation.StructConstantStoredProperty
[ RUN      ] ForwardModeDifferentiation.StructMutation
[       OK ] ForwardModeDifferentiation.StructMutation
[ RUN      ] ForwardModeDifferentiation.StructGeneric
[       OK ] ForwardModeDifferentiation.StructGeneric
[ RUN      ] ForwardModeDifferentiation.SubsetIndices
[       OK ] ForwardModeDifferentiation.SubsetIndices
[ RUN      ] ForwardModeDifferentiation.ForceUnwrapping
[       OK ] ForwardModeDifferentiation.ForceUnwrapping
[ RUN      ] ForwardModeDifferentiation.Array.differentiableMap
stdout>>> check failed at /Users/danielzheng/swift-build/swift/test/AutoDiff/validation-test/forward_mode.swift, line 1333
stdout>>> expected: [3.0, 3.0, 3.0] (of type (extension in _Differentiation):Swift.Array<Swift.Float>.DifferentiableView)
stdout>>> actual: [3.0, 6.0, 9.0] (of type (extension in _Differentiation):Swift.Array<Swift.Float>.DifferentiableView)
[     FAIL ] ForwardModeDifferentiation.Array.differentiableMap
ForwardModeDifferentiation: Some tests failed, aborting
UXPASS: []
FAIL: ["Array.differentiableMap"]
SKIP: []
To debug, run:
$ /Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/test-macosx-x86_64/AutoDiff/validation-test/Output/forward_mode.swift.tmp/a.out --stdlib-unittest-in-process --stdlib-unittest-filter "Array.differentiableMap"

--
Command Output (stderr):
--
/Users/danielzheng/swift-build/swift/test/AutoDiff/validation-test/forward_mode.swift:272:11: warning: variable 'tuple' was never mutated; consider changing to 'let' constant
      var tuple = (2 * x, x)
      ~~~ ^
      let
/Users/danielzheng/swift-build/swift/test/AutoDiff/validation-test/forward_mode.swift:283:11: warning: variable 'tuple' was never mutated; consider changing to 'let' constant
      var tuple = (2 * x, 1)
      ~~~ ^
      let
/Users/danielzheng/swift-build/swift/test/AutoDiff/validation-test/forward_mode.swift:317:9: warning: variable 'tuple' was never mutated; consider changing to 'let' constant
    var tuple = (x, x)
    ~~~ ^
    let
/Users/danielzheng/swift-build/swift/test/AutoDiff/validation-test/forward_mode.swift:1287:9: warning: variable 'generic' was never mutated; consider changing to 'let' constant
    var generic = Generic(x: input, y: input, z: input)
    ~~~ ^
    let
/Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/test-macosx-x86_64/AutoDiff/validation-test/Output/forward_mode.swift.script: line 1: 46561 Abort trap: 6           /usr/bin/env DYLD_LIBRARY_PATH='/Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/lib/swift/macosx' LD_LIBRARY_PATH='/Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/lib/swift/macosx:' SIMCTL_CHILD_DYLD_LIBRARY_PATH='/Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/lib/swift/macosx' /Users/danielzheng/swift-build/build/Ninja-ReleaseAssert/swift-macosx-x86_64/test-macosx-x86_64/AutoDiff/validation-test/Output/forward_mode.swift.tmp/a.out
```

</details>

I wonder if you see something similar? I almost always run `lit.py -vvv` for verbose error messages.

Here's the relevant part of the error message, which is an `expectEqual` check failure at runtime:
```
[ RUN      ] ForwardModeDifferentiation.Array.differentiableMap
stdout>>> check failed at /Users/danielzheng/swift-build/swift/test/AutoDiff/validation-test/forward_mode.swift, line 1333
stdout>>> expected: [3.0, 3.0, 3.0] (of type (extension in _Differentiation):Swift.Array<Swift.Float>.DifferentiableView)
stdout>>> actual: [3.0, 6.0, 9.0] (of type (extension in _Differentiation):Swift.Array<Swift.Float>.DifferentiableView)
```

The expected result is `[3, 6, 9]` because the differential of `multiplyMap` is applied to `tan` (`[1, 2, 3]`).

```suggestion
  expectEqual([3, 6, 9], differential(at: [1, 1, 1], in: multiplyMap)(tan))
```

Reply:
Ah, that's right indeed. But I didn't see that in my `lit` output:
<details>

<p>

```
>  llvm-project/llvm/utils/lit/lit.py -vvv build/Ninja-ReleaseAssert/swift-linux-x86_64/test-linux-x86_64/AutoDiff --filter forward_mode
lit.py: /home/alex/swift-source/swift/test/lit.cfg:170: note: Using cmake: /usr/bin/cmake
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using swift-frontend: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swift-frontend
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using swift: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swift
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using swiftc: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swiftc
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using sil-opt: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/sil-opt
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using sil-func-extractor: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/sil-func-extractor
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using sil-llvm-gen: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/sil-llvm-gen
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using sil-nm: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/sil-nm
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using sil-passpipeline-dumper: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/sil-passpipeline-dumper
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using lldb-moduleimport-test: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/lldb-moduleimport-test
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using swift-ide-test: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swift-ide-test
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using swift-dependency-tool: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swift-dependency-tool
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using swift-syntax-test: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swift-syntax-test
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using swift-syntax-parser-test: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swift-syntax-parser-test
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using swift-reflection-dump: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swift-reflection-dump
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using swift-remoteast-test: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swift-remoteast-test
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using swift-indent: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swift-indent
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using swift-symbolgraph-extract: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swift-symbolgraph-extract
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using clang: /home/alex/swift-source/build/Ninja-ReleaseAssert/llvm-linux-x86_64/bin/clang
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using llvm-link: /home/alex/swift-source/build/Ninja-ReleaseAssert/llvm-linux-x86_64/bin/llvm-link
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using swift-llvm-opt: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swift-llvm-opt
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using llvm-profdata: /home/alex/swift-source/build/Ninja-ReleaseAssert/llvm-linux-x86_64/bin/llvm-profdata
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using llvm-cov: /home/alex/swift-source/build/Ninja-ReleaseAssert/llvm-linux-x86_64/bin/llvm-cov
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using llvm-strings: /home/alex/swift-source/build/Ninja-ReleaseAssert/llvm-linux-x86_64/bin/llvm-strings
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using FileCheck: /home/alex/swift-source/build/Ninja-ReleaseAssert/llvm-linux-x86_64/bin/FileCheck
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using llvm-dwarfdump: /home/alex/swift-source/build/Ninja-ReleaseAssert/llvm-linux-x86_64/bin/llvm-dwarfdump
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using llvm-readelf: /home/alex/swift-source/build/Ninja-ReleaseAssert/llvm-linux-x86_64/bin/llvm-readelf
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using llvm-dis: /home/alex/swift-source/build/Ninja-ReleaseAssert/llvm-linux-x86_64/bin/llvm-dis
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using llvm-nm: /home/alex/swift-source/build/Ninja-ReleaseAssert/llvm-linux-x86_64/bin/llvm-nm
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using sourcekitd-test: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/sourcekitd-test
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using complete-test: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/complete-test
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using swift-api-digester: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swift-api-digester
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using swift-refactor: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swift-refactor
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using swift-demangle-yamldump: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swift-demangle-yamldump
lit.py: /home/alex/swift-source/swift/test/lit.cfg:230: warning: couldn't find 'Benchmark_O' program, try setting BENCHMARK_O in your environment
lit.py: /home/alex/swift-source/swift/test/lit.cfg:230: warning: couldn't find 'Benchmark_Driver' program, try setting BENCHMARK_DRIVER in your environment
lit.py: /home/alex/swift-source/swift/test/lit.cfg:230: warning: couldn't find 'wasmer' program, try setting WASMER in your environment
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using wasm-ld: /home/alex/swift-source/build/Ninja-ReleaseAssert/llvm-linux-x86_64/bin/wasm-ld
lit.py: /home/alex/swift-source/swift/test/lit.cfg:304: note: Using resource dir: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/lib/swift
lit.py: /home/alex/swift-source/swift/test/lit.cfg:347: note: Compiling with -swift-version 4
lit.py: /home/alex/swift-source/swift/test/lit.cfg:370: note: Using Clang module cache: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/swift-test-results/x86_64-unknown-linux-gnu/clang-module-cache
lit.py: /home/alex/swift-source/swift/test/lit.cfg:371: note: Using test results dir: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/./swift-test-results/x86_64-unknown-linux-gnu
lit.py: /home/alex/swift-source/swift/test/lit.cfg:376: note: Using code completion cache: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/swift-test-results/x86_64-unknown-linux-gnu/completion-cache
lit.py: /home/alex/swift-source/swift/test/lit.cfg:1150: note: Testing Linux x86_64-unknown-linux-gnu
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using swift-autolink-extract: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swift-autolink-extract
lit.py: /home/alex/swift-source/swift/test/lit.cfg:1652: note: Using platform module dir: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/lib/swift/linux/x86_64
lit.py: /home/alex/swift-source/swift/test/lit.cfg:1677: note: Testing with the just-built libraries at /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/lib/swift/linux/x86_64
lit.py: /home/alex/swift-source/swift/test/lit.cfg:227: note: using swift-reflection-test-linux-x86_64: /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swift-reflection-test-linux-x86_64
lit.py: /home/alex/swift-source/swift/test/lit.cfg:1953: note: Running tests on Arch-rolling
lit.py: /home/alex/swift-source/swift/test/lit.cfg:1963: note: Available features: CMAKE_GENERATOR=Ninja, CODEGENERATOR=AArch64, CODEGENERATOR=ARM, CODEGENERATOR=Mips, CODEGENERATOR=PowerPC, CODEGENERATOR=SystemZ, CODEGENERATOR=X86, CPU=x86_64, LinuxDistribution=Arch-rolling, OS=linux-gnu, PTRSIZE=64, SWIFT_VERSION=4, VENDOR=unknown, asan_runtime, asserts, autolink-extract, crash-recovery, differentiable_programming, executable_test, fuzzer_runtime, libdispatch, no_asan, no_lto, nonexecutable_test, optimized_stdlib, optimized_stdlib_x86_64, runtime_function_counters, scudo_runtime, shell, sourcekit, standalone_build, swift-remoteast-test, swift_ast_verifier, swift_interpreter, swift_only_stable_abi, swift_stdlib_asserts, swift_test_mode_optimize_none, swift_test_mode_optimize_none_x86_64, swift_tools_extra, syntax_parser_lib, tsan_runtime, ubsan_runtime
-- Testing: 2 of 105 tests, 2 workers --
PASS: Swift(linux-x86_64) :: AutoDiff/SILOptimizer/forward_mode_diagnostics.swift (1 of 2)
FAIL: Swift(linux-x86_64) :: AutoDiff/validation-test/forward_mode.swift (2 of 2)
******************** TEST 'Swift(linux-x86_64) :: AutoDiff/validation-test/forward_mode.swift' FAILED ********************
Script:
--
: 'RUN: at line 1';   rm -rf "/home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/test-linux-x86_64/AutoDiff/validation-test/Output/forward_mode.swift.tmp" && mkdir -p "/home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/test-linux-x86_64/AutoDiff/validation-test/Output/forward_mode.swift.tmp" && /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swiftc -target x86_64-unknown-linux-gnu -toolchain-stdlib-rpath  -module-cache-path '/home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/swift-test-results/x86_64-unknown-linux-gnu/clang-module-cache' -swift-version 4  -Xfrontend -ignore-module-source-info  -module-cache-path '/home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/swift-test-results/x86_64-unknown-linux-gnu/clang-module-cache' /home/alex/swift-source/swift/test/AutoDiff/validation-test/forward_mode.swift -Xfrontend -enable-experimental-forward-mode-differentiation -o /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/test-linux-x86_64/AutoDiff/validation-test/Output/forward_mode.swift.tmp/a.out -module-name main  && echo /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/test-linux-x86_64/AutoDiff/validation-test/Output/forward_mode.swift.tmp/a.out && /usr/bin/env DYLD_LIBRARY_PATH='/home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/lib/swift/linux/x86_64' LD_LIBRARY_PATH='/home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/lib/swift/linux/x86_64:/home/alex/swift-source/build/Ninja-ReleaseAssert/libdispatch-linux-x86_64' SIMCTL_CHILD_DYLD_LIBRARY_PATH='/home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/lib/swift/linux/x86_64'  /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/test-linux-x86_64/AutoDiff/validation-test/Output/forward_mode.swift.tmp/a.out
--
Exit Code: 134

Command Output (stdout):
--
/home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/test-linux-x86_64/AutoDiff/validation-test/Output/forward_mode.swift.tmp/a.out
[ RUN      ] ForwardModeDifferentiation.Identity
[       OK ] ForwardModeDifferentiation.Identity
[ RUN      ] ForwardModeDifferentiation.Unary
[       OK ] ForwardModeDifferentiation.Unary
[ RUN      ] ForwardModeDifferentiation.Binary
[       OK ] ForwardModeDifferentiation.Binary
[ RUN      ] ForwardModeDifferentiation.BinaryWithLets
[       OK ] ForwardModeDifferentiation.BinaryWithLets
[ RUN      ] ForwardModeDifferentiation.SubsetParametersDiff
[       OK ] ForwardModeDifferentiation.SubsetParametersDiff
[ RUN      ] ForwardModeDifferentiation.UnaryWithVars
[       OK ] ForwardModeDifferentiation.UnaryWithVars
[ RUN      ] ForwardModeDifferentiation.StructInit
[       OK ] ForwardModeDifferentiation.StructInit
[ RUN      ] ForwardModeDifferentiation.StructExtract
[       OK ] ForwardModeDifferentiation.StructExtract
[ RUN      ] ForwardModeDifferentiation.LocalStructVariable
[       OK ] ForwardModeDifferentiation.LocalStructVariable
[ RUN      ] ForwardModeDifferentiation.noParamMethodA
[       OK ] ForwardModeDifferentiation.noParamMethodA
[ RUN      ] ForwardModeDifferentiation.noParamMethodx
[       OK ] ForwardModeDifferentiation.noParamMethodx
[ RUN      ] ForwardModeDifferentiation.complexBinaryMethod
[       OK ] ForwardModeDifferentiation.complexBinaryMethod
[ RUN      ] ForwardModeDifferentiation.TrackedIdentity
[       OK ] ForwardModeDifferentiation.TrackedIdentity
[ RUN      ] ForwardModeDifferentiation.TrackedAddition
[       OK ] ForwardModeDifferentiation.TrackedAddition
[ RUN      ] ForwardModeDifferentiation.TrackedDivision
[       OK ] ForwardModeDifferentiation.TrackedDivision
[ RUN      ] ForwardModeDifferentiation.TrackedMultipleMultiplication
[       OK ] ForwardModeDifferentiation.TrackedMultipleMultiplication
[ RUN      ] ForwardModeDifferentiation.TrackedWithLets
[       OK ] ForwardModeDifferentiation.TrackedWithLets
[ RUN      ] ForwardModeDifferentiation.TupleLet
[       OK ] ForwardModeDifferentiation.TupleLet
[ RUN      ] ForwardModeDifferentiation.TupleVar
[       OK ] ForwardModeDifferentiation.TupleVar
[ RUN      ] ForwardModeDifferentiation.TupleMutation
[       OK ] ForwardModeDifferentiation.TupleMutation
[ RUN      ] ForwardModeDifferentiation.TupleNonDifferentiableElements
[       OK ] ForwardModeDifferentiation.TupleNonDifferentiableElements
[ RUN      ] ForwardModeDifferentiation.GenericIdentity
[       OK ] ForwardModeDifferentiation.GenericIdentity
[ RUN      ] ForwardModeDifferentiation.GenericTensorIdentity
[       OK ] ForwardModeDifferentiation.GenericTensorIdentity
[ RUN      ] ForwardModeDifferentiation.GenericTensorPlus
[       OK ] ForwardModeDifferentiation.GenericTensorPlusDetails
[ RUN      ] ForwardModeDifferentiation.GenericTensorBinaryInput
[       OK ] ForwardModeDifferentiation.GenericTensorBinaryInput
[ RUN      ] ForwardModeDifferentiation.GenericTensorWithLets
[       OK ] ForwardModeDifferentiation.GenericTensorWithLets
[ RUN      ] ForwardModeDifferentiation.GenericTensorWithVars
[       OK ] ForwardModeDifferentiation.GenericTensorWithVars
[ RUN      ] ForwardModeDifferentiation.GenericTrackedIdentity
[       OK ] ForwardModeDifferentiation.GenericTrackedIdentity
[ RUN      ] ForwardModeDifferentiation.GenericTrackedBinaryAdd
[       OK ] ForwardModeDifferentiation.GenericTrackedBinaryAdd
[ RUN      ] ForwardModeDifferentiation.GenericTrackedBinaryLets
[       OK ] ForwardModeDifferentiation.GenericTrackedBinaryLets
[ RUN      ] ForwardModeDifferentiation.GenericTrackedBinaryVars
[       OK ] ForwardModeDifferentiation.GenericTrackedBinaryVars
[ RUN      ] ForwardModeDifferentiation.TrackedDifferentiableFuncType
[       OK ] ForwardModeDifferentiation.TrackedDifferentiableFuncType
[ RUN      ] ForwardModeDifferentiation.Final
[       OK ] ForwardModeDifferentiation.Final
[ RUN      ] ForwardModeDifferentiation.Simple
[       OK ] ForwardModeDifferentiation.Simple
[ RUN      ] ForwardModeDifferentiation.SimpleWrtSelf
[       OK ] ForwardModeDifferentiation.SimpleWrtSelf
[ RUN      ] ForwardModeDifferentiation.Simple Protocol
[       OK ] ForwardModeDifferenti
--
Command Output (stderr):
--
/home/alex/swift-source/swift/test/AutoDiff/validation-test/forward_mode.swift:272:11: warning: variable 'tuple' was never mutated; consider changing to 'let' constant
      var tuple = (2 * x, x)
      ~~~ ^
      let
/home/alex/swift-source/swift/test/AutoDiff/validation-test/forward_mode.swift:283:11: warning: variable 'tuple' was never mutated; consider changing to 'let' constant
      var tuple = (2 * x, 1)
      ~~~ ^
      let
/home/alex/swift-source/swift/test/AutoDiff/validation-test/forward_mode.swift:317:9: warning: variable 'tuple' was never mutated; consider changing to 'let' constant
    var tuple = (x, x)
    ~~~ ^
    let
/home/alex/swift-source/swift/test/AutoDiff/validation-test/forward_mode.swift:1287:9: warning: variable 'generic' was never mutated; consider changing to 'let' constant
    var generic = Generic(x: input, y: input, z: input)
    ~~~ ^
    let
/home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/test-linux-x86_64/AutoDiff/validation-test/Output/forward_mode.swift.script: line 1: 20950 Aborted                 (core dumped) /usr/bin/env DYLD_LIBRARY_PATH='/home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/lib/swift/linux/x86_64' LD_LIBRARY_PATH='/home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/lib/swift/linux/x86_64:/home/alex/swift-source/build/Ninja-ReleaseAssert/libdispatch-linux-x86_64' SIMCTL_CHILD_DYLD_LIBRARY_PATH='/home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/lib/swift/linux/x86_64' /home/alex/swift-source/build/Ninja-ReleaseAssert/swift-linux-x86_64/test-linux-x86_64/AutoDiff/validation-test/Output/forward_mode.swift.tmp/a.out

--

********************

Testing Time: 6.82s
********************
Failing Tests (1):
    Swift(linux-x86_64) :: AutoDiff/validation-test/forward_mode.swift

  Expected Passes    : 1
  Unexpected Failures: 1

3 warning(s) in tests 
```

</p>

</details>

Maybe some issue with my build.

I fixed the bug and the test is passing for me now.

Reply:
Huh, that's weird. It seems like `stdout` output is cut off for some reason, maybe that's when the crash occurs:
```
[ RUN      ] ForwardModeDifferentiation.Simple Protocol
[       OK ] ForwardModeDifferenti # cut off here
```

I don't think I've encountered this before.